### PR TITLE
refactor: move to yargs, use chalk

### LINKED ISSUE #123: Why not switching over to a better argument parser library?
**What problem do you want to solve?**
The project as of now makes use of `minimist` for the purpose of argument parsing. One can come across a single source file with bunch of `if-else` statements essentially affecting the DX.

**What do you think is the correct solution to this problem?**
Switch over to a better `argument parser library` which does the job. Thereby we can adopt the principles of modularity making the code-base a lot cleaner. If bundle size is worth concern, [commander.js](https://github.com/tj/commander.js/) would be a great choice .

**Are you willing to submit a pull request to implement this change?**
Ofc :+1:

### LINKED ISSUE #141: Allow providing additional options to MPV
MPV has some useful options like "--no-video" (for mp3 files) and "--full-screen" which can be used while streaming. Please add support for passing additional options to the media player.

For example, a magnet link to an mp3 file can be streamed as follows, without opening a window.
```
webtorrent "magnet:...." --mpv -- --no-video
```
I borrowed the syntax for mpv options from peerflix.

### LINKED ISSUE #143: Stderr maxBuffer length exceeded
**What version of this package are you using?**
3.0.7 (0.108.6)

**What operating system, Node.js, and npm version?**
Windows 10 Enterprise (Build 2004), npm 6.14.6, nodejs v12.18.3

**What happened?**
Was testing out a large video stream to see if this issue still persists and it still does. The torrent file size is 21gb and this error was thrown at 15gb/21gb. 

*The error:*

    Error: stderr maxBuffer length exceeded

*The input:*

    webtorrent download "<MAGNET_URL>" -o Downloads --vlc --not-on-top

**What did you expect to happen?**
Expected the stream to continue without throwing an error.

**Are you willing to submit a pull request to fix this bug?**
Maybe

### LINKED ISSUE #149: Busy port detection
**What version of this package are you using?**
```
webttorrent-cli --version  
3.0.7 (0.108.6)
```

**What operating system, Node.js, and npm version?**
```
Debian Sid
node --version                        
v12.19.0
```

**What happened?**

If I start and the port is used by some other application webtorrent-cli start twice the media player. For example if I use port 8000 (the default streaming port) with `python3 -m http.server` and try to start `webtorrent-cli /link/path --vlc -t /subtitles/path -s 16` VLC is tarted twice. If I close one of the VLC players webttorrent closes immediately, no matters which VLC instance is closed. The very same behavior is observed with mpv/mplayer. I'm aware with port option in cli but still thinks this is a bug.

### PR DESCRIPTION
**Pull request by OsmanAltun**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update
[x] Bug fix
[x] New feature
[x] Other, please explain:

Project revamp. I tried both CommanderJS and Yargs. I also tried making a custom wrapper for minimist, however it took too much effort. I decided to stick with Yargs.

**What changes did you make? (Give an overview)**
Lots.
1. OpenPlayer function now uses child_process.spawn instead of child_process.exec.
2. Removed the openIINA function which was a weird function. 
3. Merged PLAYER_ARGS with PLAYER_EXEC into playerArgs. 
4. replaced Minimist with Yargs. 
5. Created init function. This function contains some if else statements which used to be on the root. 
6. Cleanup of dependencies and added comments.
7. Feature: Ability to give player specific arguments.
8. Updated README.md file.
9. If the given port is busy, it will pick a random port.
10. Replace Clivas with the combined use of Chalk and console.log. 

**Which issue (if any) does this pull request address?**
 - closes #143
 - closes #123
 - closes #141 
 - closes #149 
 - closes #90 

**Is there anything you'd like reviewers to focus on?**
Refer to changes.

### CODE REVIEW

**Comment by OsmanAltun** [comment]
Small status update:

Waiting for Yargs to merge my PR's, so that the hacky workarounds can be removed.

**Comment by DiegoRBaquero** [comment]
I feel this PR could be broken down into smaller ones, hard to review so many changes

**Comment by OsmanAltun** [comment]
I think I should rebase first before anything (haven't looked into this in a while).

File: README.md [hunk 1]
```diff
@@ -65,54 +65,65 @@ $ webtorrent --help
    \_/\_/ \___|_.__/ \__\___/|_|  |_|  \___|_| |_|\__|
 
 Usage:
-  webtorrent [command] <torrent-id> <options>
+  webtorrent [command] <torrent-id> [options]
 
-Example:
+Examples:
   webtorrent download "magnet:..." --vlc
+  webtorrent "magnet:..." --vlc --player-args="--video-on-top --repeat"
```
Comment (DiegoRBaquero) [inline README.md:72]: This doesn't match any command though

File: .gitignore [hunk 2]
```diff
@@ -1 +1,168 @@
-node_modules
+
```
Comment (DiegoRBaquero) [inline .gitignore:1]: I'd rather keep the lean one, if this PR requires extra ignore, please add those specifically

File: bin/cmd.js [hunk 3]
```diff
@@ -1,334 +1,243 @@
 #!/usr/bin/env node
 'use strict'
 
-const clivas = require('clivas')
+// #region Variables
```
Comment (DiegoRBaquero) [inline bin/cmd.js:4]: ?

File: bin/cmd.js [hunk 4]
```diff
@@ -828,20 +695,20 @@ function drawTorrent (torrent) {
...
-        prettierBytes(wire.uploadSpeed())
+          : 'Unknown').padEnd(25),
+        prettierBytes(wire.downloaded).padEnd(10),
+        (prettierBytes(wire.downloadSpeed()) + '/s').padEnd(12),
+        (prettierBytes(wire.uploadSpeed()) + '/s').padEnd(12)
       ]
 
       if (argv.verbose) {
-        str += ' {15+grey:%s} {10+grey:%s}'
+        str += chalk` {grey %s} {grey %s}`
```
Comment (DiegoRBaquero) [inline bin/cmd.js:711]: I know this syntax requires a min node version, which is it?

**Review by DiegoRBaquero — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (OsmanAltun) [reply README.md:72]: `{ command: '$0 [torrent-ids...]', handler: (args) => { processInputs(args.torrentIds, runDownload) } },`

this ? Line ~73

File: bin/cmd.js (hunk 4 above)
Comment (OsmanAltun) [reply bin/cmd.js:701]: chalk requires >= 12.17 (LTS). 
If u meant tagged template literals, they are supported in all nodejs versions I believe (check ECMAscript support matrix).

File: bin/cmd.js (hunk 3 above)
Comment (OsmanAltun) [reply bin/cmd.js:4]: Replaced clivas with chalk+console.log. Clivas is old and broken, it caused the terminal to print the last line multiple times (if 
 I remember correctly). Chalk is well maintained and `console.log` is native 😃

File: .gitignore (hunk 2 above)
Comment (OsmanAltun) [reply .gitignore:1]: alright

File: README.md (hunk 1 above)
Comment (DiegoRBaquero) [reply README.md:72]: I mean, it's not in the readme/help command

File: bin/cmd.js (hunk 3 above)
Comment (DiegoRBaquero) [reply bin/cmd.js:4]: I'm wondering about what #region is

File: bin/cmd.js (hunk 4 above)
Comment (DiegoRBaquero) [reply bin/cmd.js:701]: awesome!

File: .gitignore [hunk 5]
```diff
@@ -1 +1,2 @@
 node_modules
+.DS_Store
```
Comment (DiegoRBaquero) [inline .gitignore:2]: newline at EOF

File: bin/cmd.js (hunk 3 above)
Comment (OsmanAltun) [reply bin/cmd.js:4]: Oh, lol. It allowes me to collapse parts in vscode.

File: README.md (hunk 1 above)
Comment (OsmanAltun) [reply README.md:72]: this implies default command:
```
Usage:
  webtorrent [command] <torrent-id> [options]
```

it is possible to add default command description to commands section of the help output (extra entry).
I can also add `[default]` to the description of `download`

Comment (DiegoRBaquero) [reply README.md:72]: yup, that'd work

File: .gitignore (hunk 5 above)
Comment (DiegoRBaquero) [inline .gitignore:2]: Removing as this should be globally ignored locally

**Review by DiegoRBaquero — APPROVED** [review]
Code LGTM, need to test app

**Comment by pc00per** [comment]
Is there anyway to use `--ytdl-format=` & `--audio-channels=` ?

**Pull request merged**